multichain-testing/test/fast-usdc/fast-usdc.test.ts

```diff
@@ -493,6 +566,221 @@ test.serial('distribute FastUSDC contract fees', async t => {
   t.truthy(balance?.amount);
 });
 
-test.todo('insufficient LP funds; forward path');
+test.serial('insufficient LP funds; forward path', async t => {
+  const eudChain = 'osmosis';
+  const mintAmt = LP_DEPOSIT_AMOUNT * 2n;
+  const {
+    assertTxStatus,
+    getUsdcDenom,
+    makeFakeEvidence,
+    nobleTools,
+    nobleAgoricChannelId,
+    retryUntilCondition,
+    txOracles,
+    useChain,
+    vstorageClient,
+  } = t.context;
+
+  // EUD wallet on the specified chain
+  const eudWallet = await createWallet(useChain(eudChain).chain.bech32_prefix);
+  const EUD = (await eudWallet.getAccounts())[0].address;
+  t.log(`EUD wallet created: ${EUD}`);
+
+  // parameterize agoric address
+  const { settlementAccount } = await vstorageClient.queryData(
+    `published.${contractName}`,
+  );
+  t.log('settlementAccount address', settlementAccount);
+
+  const recipientAddress = encodeAddressHook(settlementAccount, { EUD });
+  t.log('recipientAddress', recipientAddress);
+
+  // register forwarding address on noble
+  const txRes = nobleTools.registerForwardingAcct(
+    nobleAgoricChannelId,
+    recipientAddress,
+  );
+  t.is(txRes?.code, 0, 'registered forwarding account');
+
+  const { address: userForwardingAddr } = nobleTools.queryForwardingAddress(
+    nobleAgoricChannelId,
+    recipientAddress,
+  );
+  t.log('got forwardingAddress', userForwardingAddr);
+
+  const evidence = makeFakeEvidence(
+    mintAmt,
+    userForwardingAddr,
+    recipientAddress,
+  );
+
+  t.log('User initiates EVM burn:', evidence.txHash);
+  // submit evidences
+  await Promise.all(txOracles.map(async o => o.submit(evidence)));
+
+  const queryClient = makeQueryClient(
+    await useChain(eudChain).getRestEndpoint(),
+  );
+
+  await assertTxStatus(evidence.txHash, 'OBSERVED');
+
+  nobleTools.mockCctpMint(mintAmt, userForwardingAddr);
+
+  await assertTxStatus(evidence.txHash, 'FORWARDED');
+  await t.notThrowsAsync(async () => {
+    await retryUntilCondition(
+      () => queryClient.queryBalance(EUD, getUsdcDenom(eudChain)),
+      ({ balance }) => {
+        if (!balance) return false; // retry
+        const value = BigInt(balance.amount);
+        if (value === 0n) return false; // retry
+        if (value < mintAmt) {
+          throw Error(`fees were deducted: ${value} < ${mintAmt}`);
+        }
+        t.log('forward done', value, 'uusdc');
+        return true;
+      },
+      `${EUD} forward available from fast-usdc`,
+      // this resolves quickly, so _decrease_ the interval so the timing is more apparent
+      { retryIntervalMs: 500, maxRetries: 20 },
+    );
+  });
```

Comment:
Appreciate the refactors to reusable helpers in test context 👏 . This could be another candidate